staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -128,6 +128,16 @@ func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) fie
 	return allErrs
 }
 
+// We don't know how to handle mutated selector for this unexpected group version.
+// Throw an internal error.
+func ValidateImmutableFieldForUnexpectedGroupVersion(newVal, oldVal interface{}, fldPath *field.Path, groupVersion string) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(ValidateImmutableField(newVal, oldVal, fldPath)) > 0 {
```

Comment:
don't make this conditional on the field being mutated... it's a programming error to not make a decision about this and we don't want to ship a release with an unknown version in these cases

Reply:
Is there a way to catch this programming error, when a new apiversion is added without making a decision in update strategy?

Reply:
a unit test in the registry packages to check registered versions for the relevant objects would be my recommendation.

also, this isn't generic enough to warrant being in k8s.io/apimachinery

Reply:
> a unit test in the registry packages to check registered versions for the relevant objects would be my recommendation.

Did you mean a unit test here, or something else?
https://github.com/kubernetes/kubernetes/blob/3a15fdbe7e1ceefa59d7cc41ac6d65998cef1b1a/pkg/registry/extensions/daemonset/strategy_test.go#L47-L73

If a new version is added without adding another test case here, the error still cannot be caught.